src/ssl.c

```diff
@@ -752,6 +752,20 @@ static int meth_sni(lua_State *L)
   if (pctx->mode == LSEC_MODE_CLIENT) {
     name = luaL_checkstring(L, 2);
     SSL_set_tlsext_host_name(ssl->ssl, name);
+
+#ifdef X509_CHECK_FLAG_NO_PARTIAL_WILDCARDS
+    /* Validate the hostname against the certificate. */
+
+    /* This is only available in OpenSSL >= 1.0.2, hence the ifdef. (Really */
+    /* we're interested in X509_VERIFY_PARAM_set1_host, more than the flag, */
+    /* but both were introduced in the same release.) On older versions the */
+    /* functionality would have to be implemented fully manually. */
+    X509_VERIFY_PARAM *param = SSL_get0_param(ssl->ssl);
+    X509_VERIFY_PARAM_set_hostflags(param,
```

Comment:
Note that this only suppresses partial wildcards (like `foo*bar.example.com`), not normal wildcards (like `*.example.com`). Partial wildcards are practically never seen in the wild (and for our case in particular , so I thought this would be a prudent safety measure. I also found this reference in a rejected issue for the Go standard library, saying that the issue was rejected due to consensus from the CA/Browser Forum that partial wildcards were a bad idea: https://github.com/golang/go/issues/27629#issuecomment-420428194 — if Go can afford to not support them then I think we can as well.

Reply:
I still don't understand why we bother to add this line if we don't need it. Why no keep the code simpler and stick with our requirements?

Reply:
Like said, I think it's a prudent safety measure. In case we ever run into such a certificate, we most likely don't want to accept it.

But I remembered another reason now, which is that this is what libwebsockets also does:

https://github.com/warmcat/libwebsockets/blob/6d8039c13b80f530edecd833eda608175ce15387/lib/tls/openssl/openssl-client.c#L184-L186

And since we're also using that, this way we're compatible in the way we treat TLS certificates.

Reply:
Sure, please add all these explanations to the commit message. Also, I don't think it is necessary to leave any comments in the code, you could add everything to the commit message and let git blame do its thing.